contracts/Depository.sol

```diff
@@ -344,58 +358,39 @@ contract Depository is IErrorsTokenomics {
                 revert OwnerOnly(msg.sender, mapUserBonds[bondIds[i]].account);
             }
 
-            // Get the productId for its status check
-            uint256 productId = mapUserBonds[bondIds[i]].productId;
-
-            // Close the program if it was not yet closed
-            if (mapBondProducts[productId].expiry > 0) {
-                uint256 supply = mapBondProducts[productId].supply;
-                // Refund unused OLAS supply from the program if not used completely
-                if (supply > 0) {
-                    ITokenomics(tokenomics).refundFromBondProgram(supply);
-                }
-                address token = mapBondProducts[productId].token;
-                delete mapBondProducts[productId];
-
-                emit CloseProduct(token, productId);
-            }
             // Increase the payout
             payout += pay;
+
+            // Get the productId
+            uint256 productId = mapUserBonds[bondIds[i]].productId;
+
             // Delete the Bond struct and release the gas
             delete mapUserBonds[bondIds[i]];
-            emit RedeemBond(productId, msg.sender, bondIds[i]);
+            emit RedeemBond(productId, msg.sender, bondIds[i], block.timestamp);
         }
 
         // Check for the non-zero payout
         if (payout == 0) {
             revert ZeroValue();
         }
+
         // No reentrancy risk here since it's the last operation, and originated from the OLAS token
         // No need to check for the return value, since it either reverts or returns true, see the ERC20 implementation
         IToken(olas).transfer(msg.sender, payout);
     }
 
-    /// @dev Gets an array of active or inactive (but not deleted) product Ids for a specific token.
-    /// @param active Flag to select active or inactive products.
+    /// @dev Gets an array of active product Ids for a specific token.
     /// @return productIds Product Ids.
-    function getProducts(bool active) external view returns (uint256[] memory productIds) {
+    function getProducts() external view returns (uint256[] memory productIds) {
         // Calculate the number of active products
         uint256 numProducts = productCounter;
         bool[] memory positions = new bool[](numProducts);
         uint256 numSelectedProducts;
-        if (active) {
-            for (uint256 i = 0; i < numProducts; i++) {
-                if (mapBondProducts[i].supply > 0 && mapBondProducts[i].expiry > block.timestamp) {
-                    positions[i] = true;
-                    ++numSelectedProducts;
-                }
-            }
-        } else {
-            for (uint256 i = 0; i < numProducts; i++) {
-                if (mapBondProducts[i].token != address(0) && mapBondProducts[i].expiry <= block.timestamp) {
-                    positions[i] = true;
-                    ++numSelectedProducts;
-                }
```

Comment:
There are no inactive products anymore, as they are either active when the supply is not zero, or closed ones, and all the product data is cleared.

Reply:
Hm ... inactive products equivalent are now those with supply zero... might be worth keeping the old format and changing the meaning of active/inactive?

Reply:
I don't see a scenario to distinguish "inactive" and "non-exists".
It seems to me that the new approach makes existence state "inactive" impossible - if we leave the garbage collection.
Which product will be considered "inactive" but will remain in storage?

Reply:
Once the supply hits zero, the product is closed. This way things are more straightforward in implementation and there are no additional branches. For historical reasons one can read from events.

Reply:
Just pointing out that a product that once got filled (zero remaining supply) is different from one that never existed. I think it's a design smell that we don't have different names for these different products.

Reply:
Let's revert the classification, just to get the product id numbers that are already inactive.